sdk/core/azure-core/src/main/java/com/azure/core/util/IOUtils.java

```diff
@@ -119,4 +122,51 @@ public void failed(Throwable e, ByteBuffer attachment) {
             sink.success();
         }
     }
+
+    /**
+     * Transfers the {@link StreamResponse} content to {@link AsynchronousByteChannel}.
+     * Resumes the transfer in case of errors.
+     *
+     * @param channel The destination {@link AsynchronousByteChannel}.
+     * @param initialResponse The initial {@link StreamResponse}.
+     * @param onErrorResume A {@link BiFunction} of {@link Throwable} and {@link Long} which is used to resume
+     * downloading when an error occurs. The function accepts a {@link Throwable} and offset at the destination
+     * from beginning of writing at which the error occurred.
+     * @param progressReporter The {@link ProgressReporter}.
+     * @param maxRetries The maximum number of times a download can be resumed when an error occurs.
+     * @return A {@link Mono} which completion indicates successful transfer.
+     */
+    public static Mono<Void> transferStreamResponseToAsynchronousByteChannel(
+        AsynchronousByteChannel channel,
+        StreamResponse initialResponse,
+        BiFunction<Throwable, Long, Mono<StreamResponse>> onErrorResume,
+        ProgressReporter progressReporter, int maxRetries) {
+
+        return transferStreamResponseToAsynchronousByteChannelHelper(
+            new ByteCountingAsynchronousByteChannel(channel, null, progressReporter),
+            initialResponse, onErrorResume, maxRetries, 0);
+    }
+
+    private static Mono<Void> transferStreamResponseToAsynchronousByteChannelHelper(
+        ByteCountingAsynchronousByteChannel channel,
+        StreamResponse response,
+        BiFunction<Throwable, Long, Mono<StreamResponse>> onErrorResume,
+        int maxRetries, int retryCount) {
+
+        return response.writeValueToAsync(channel)
+            .doFinally(ignored -> response.close())
+            .onErrorResume(Exception.class, exception -> {
+
+                int updatedRetryCount = retryCount + 1;
+
+                if (updatedRetryCount > maxRetries) {
+                    return Mono.error(exception);
```

Comment:
Add a log here to help troubleshoot what went wrong. The caller of this helper method may or may not log it.